src/TodoList.tsx

```diff
@@ -0,0 +1,106 @@
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+
+interface TodoListProps {
+  todos: Todo[],
+  removeTodo: React.Dispatch<React.SetStateAction<boolean>>,
+  editTodo: React.Dispatch<React.SetStateAction<boolean>>,
+}
+
+export const TodoList: React.FC<TodoListProps> = ({
+  todos,
+  removeTodo,
+  editTodo,
+}) => {
+  const handleTodoRemoval = () => {
+    removeTodo(true);
+  };
+
+  const handleTodoTitleChange = () => {
+    editTodo(true);
+  };
+
+  return (
+    <section className="todoapp__main">
+      {/* This is a completed todo */}
+
+      {todos.map((todo) => (
+        <div
+          className={classNames('todo', {
+            completed: todo.completed,
+          })}
+          key={todo.id}
+          onDoubleClick={handleTodoTitleChange}
+        >
+          <label className="todo__status-label">
+            <input
+              type="checkbox"
+              className="todo__status"
+              checked
+            />
+          </label>
+
+          <span className="todo__title">{todo.title}</span>
+
+          {/* Remove button appears only on hover */}
+          <button
+            type="button"
+            className="todo__remove"
+            onClick={handleTodoRemoval}
+          >
+            ×
+          </button>
+
+          {/* overlay will cover the todo while it is being updated */}
+          <div className="modal overlay">
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      ))}
+
+      {/* This todo is not completed */}
+
+      {/* This todo is being edited */}
+      {/* <div className="todo">
+        <label className="todo__status-label">
+          <input
+            type="checkbox"
+            className="todo__status"
+          />
+        </label> */}
+
+      {/* This form is shown instead of the title and remove button */}
+      {/* <form>
+          <input
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value="Todo is being edited now"
+          />
+        </form>
+
+        <div className="modal overlay">
+          <div className="modal-background has-background-white-ter" />
+          <div className="loader" />
+        </div>
+      </div> */}
+
+      {/* This todo is in loadind state */}
+      {/* <div className="todo">
+        <label className="todo__status-label">
+          <input type="checkbox" className="todo__status" />
+        </label>
+
+        <span className="todo__title">Todo is being saved now</span>
+        <button type="button" className="todo__remove">×</button> */}
+
+      {/* 'is-active' class puts this modal on top of the todo */}
+      {/* <div className="modal overlay is-active">
+          <div className="modal-background has-background-white-ter" />
+          <div className="loader" />
+        </div> */}
+      {/* </div> */}
```

Comment:
Why are these comments here?)

Reply:
they are default ones, I thought I would use them later when implementing posting and patching